sktime_dl/deeplearning/base/estimators/_regressor.py

```diff
@@ -0,0 +1,88 @@
+# Base class for regressors
+
+__author__ = "Withington"
+
+import numpy as np
+import pandas as pd
+
+from sklearn.base import RegressorMixin
+from sklearn.utils.validation import check_is_fitted
+
+from sktime.regressors.base import BaseRegressor
+from sktime.utils.validation.supervised import validate_X, validate_X_y
+
+from sktime_dl.utils import save_trained_model
+
+
+class BaseDeepRegressor(BaseRegressor, RegressorMixin):
+    model_save_directory = None
```

Comment:
Why do we want the `model_save_directory` as a class attribute?

Reply:
```BaseDeepClassifier``` already had ```model_save_directory``` as a class attribute so it is included in ```BaseDeepRegressor``` too. 

```self.save_trained_model()``` is always called at the end of ```fit(X, y)``` and the model is saved if ```model_save_directory``` is not ```None```.

It is also used occasionally in callbacks, e.g. ResNet and TLeNet ```keras.callbacks.ModelCheckpoint```

I wouldn't put ```model_save_directory``` in ```BaseDeepNetwork``` because a network is incomplete, it's not a full model.

Does that answer your question or did you have something else in mind?

Reply:
I'm not sure why it's defined on the class and not the instance level?

Reply:
Good question. Perhaps all ```BaseDeepRegressor``` and ```BaseDeepClassifier``` class attributes need to be moved to the instance level? Currently we have:

```
class BaseDeepClassifier(BaseClassifier):
    classes_ = None
    nb_classes = None
    model_save_directory = None
    model = None
    model_name = None
```

Any reason why it would need to stay like this? Memory considerations? @James-Large ?

I'll try it locally and test it.

Reply:
Class attributes moved to instance attributes and ```__init__``` calls updated accordingly. What do you make of it?

Reply:
Looks good to me, happy to merge once @James-Large has reviewed it! :-)

Reply:
This makes more sense for sure, problem of poor python-esque code understanding back then on my part probably